.changeset/tidy-seas-sing.md

```diff
@@ -0,0 +1,79 @@
+---
+"thirdweb": patch
+---
+
+Added Bridge.Onramp.prepare and Bridge.Onramp.status functions
+
+## Bridge.Onramp.prepare
+
+Prepares an onramp transaction, returning a link from the specified provider to onramp to the specified token.
+
+```typescript
+import { Bridge } from "thirdweb";
+import { ethereum } from "thirdweb/chains";
+import { NATIVE_TOKEN_ADDRESS, toWei } from "thirdweb/utils";
+
+const preparedOnramp = await Bridge.Onramp.prepare({
+  client: thirdwebClient,
+  onramp: "stripe",
+  chainId: ethereum.id,
+  tokenAddress: NATIVE_TOKEN_ADDRESS,
+  receiver: "0x...", // receiver's address
+  amount: toWei("10"), // 10 of the destination token
+  // Optional params:
+  // sender: "0x...", // sender's address
+  // onrampTokenAddress: NATIVE_TOKEN_ADDRESS, // token to initially onramp to
+  // onrampChainId: 1, // chain to initially onramp to
+  // currency: "USD",
+  // maxSteps: 2,
+  // purchaseData: { customId: "123" }
+});
+
+console.log(preparedOnramp.link); // URL to redirect the user to
+console.log(preparedOnramp.destinationAmount); // Amount of token user will receive
```

Comment:
Instead of destination amount (since they inputted that) I would highlight the currencyAmount, how much they'll pay

Reply:
done